dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,293 @@
+        self,
+        X: npt.ArrayLike,
+        y: npt.ArrayLike,
+        *,
+        alpha: typing.Optional[float] = None,
+        num_features: typing.Optional[int] = None,
+        time_limit: typing.Optional[float] = None,
+    ) -> SelectFromQuadraticModel:
+        """Select the features to keep.
+
+        Args:
+            X:
+                Feature vectors formatted as a numerical 2D array-like.
+            y:
+                Class labels formatted as a numerical 1D array-like.
+            alpha:
+                Hyperparameter between 0 and 1 that controls the relative weight of
+                the relevance and redundancy terms.  `alpha=1` places all weight on
+                relevance and selects all features, whereas `alpha=0` places all
+                weight on redundancy and selects no features.
+                Defaults to the value provided to the constructor.
+            num_features:
+                The number of features to select.
+                Defaults to the value provided to the constructor.
+            time_limit:
+                The time limit for the run on the hybrid solver.
+                Defaults to the value provided to the constructor.
+
+        Returns:
+            This instance of `SelectFromQuadraticModel`.
+        """
+        X = np.atleast_2d(np.asarray(X))
+        if X.ndim != 2:
+            raise ValueError("X must be a 2-dimensional array-like")
+
+        # y is checked by the correlation method function
+
+        if alpha is None:
+            alpha = self._alpha
+        # alpha is checked by the correlation method function
+
+        if num_features is None:
+            num_features = self._num_features
+        # num_features is checked by the correlation method function
+
+        # time_limit is checked by the LeapHybridCQMSampelr
+
+        # if we already have fewer features than requested, just return
+        if num_features >= X.shape[1]:
+            self._mask = np.ones(X.shape[1], dtype=bool)
+            return self
+
+        if self._method == "correlation":
+            cqm = self.correlation_cqm(X, y, num_features=num_features, alpha=alpha)
+        # elif self._method == "mutual information":
+        #     cqm = self.mutual_information_cqm(X, y, num_features=num_features)
+        else:
+            raise ValueError(f"only methods {self.acceptable_methods} are implemented")
+
+        try:
+            sampler = LeapHybridCQMSampler()
+        except (ConfigFileError, SolverAuthenticationError) as e:
+            raise RuntimeError(
+                f"""Instantiation of a Leap hybrid solver failed with an {e} error.
+
+                See https://docs.ocean.dwavesys.com/en/stable/overview/sapi.html for configuring
+                access to Leap’s solvers.
+                """
+            )
+
+        sampleset = sampler.sample_cqm(cqm, time_limit=self._time_limit)
+
+        # we can probably rely on the sample set having the same variable order
```

Comment:
Is this comment actually for ``self._mask = np.fromiter ...`` ?

Reply:
It is, will remove.